crates/gazette/src/lib.rs

```diff
@@ -77,13 +77,18 @@ impl Error {
 
 pub type Result<T> = std::result::Result<T, Error>;
 
-/// Dial a gRPC endpoint with opinionated defaults and
+/// Lazily dial a gRPC endpoint with opinionated defaults and
 /// support for TLS and Unix Domain Sockets.
-pub async fn dial_channel(endpoint: &str) -> Result<tonic::transport::Channel> {
+pub fn dial_channel(endpoint: &str) -> Result<tonic::transport::Channel> {
     use std::time::Duration;
 
     let ep = tonic::transport::Endpoint::from_shared(endpoint.to_string())
         .map_err(|_err| Error::InvalidEndpoint(endpoint.to_string()))?
+        // Note this connect_timeout accounts only for TCP connection time and
+        // does not apply to time required for TLS or HTTP/2 transport start,
+        // which can block indefinitely if the server is bound but not listening.
+        // Callers MUST implement per-RPC timeouts if that's important.
```

Comment:
Pushed a small update here, which also relates to an issue we saw with a reactor connection to flow-connector-init which was ESTABLISHED even though the network namespace had been completely torn down.

Now the advice would be you don't need to worry about it, because HTTP/2 keep-alive verifies the server is up and actively serving the HTTP/2 transport. If you choose to apply a timeout on individual stream receives, you can still do so for each stream receive future.